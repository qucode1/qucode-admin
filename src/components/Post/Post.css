.title {
  font-size: 2.5rem;
  color: #2F89DE
}

.content {
  font-size: 1.6rem;
  line-height: 1.7
}

.link {
  color: #2F89DE
}

.link:focus, .link:hover {
  color: #30699E;
  transition: color .3s ease
}
