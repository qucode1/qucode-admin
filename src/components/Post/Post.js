import React from 'react'
import { Link } from 'react-router-dom'

import styles from './Post.css'

const Post = (props) => (
  <div>
    <h2 className={styles.title}>{!props.details ? <Link to={`/posts/${props.slug}`} className={styles.link}>{props.title}</Link> : props.title}</h2>
    <p className={styles.content}>{props.details ? props.content : props.excerpt}</p>
    {props.details && <Link to='/posts' className={styles.link}>Go back to Posts</Link>}
  </div>
)

export default Post
