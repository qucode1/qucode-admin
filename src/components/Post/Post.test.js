import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Post from './Post'

it('should render h2 and p with message as preview', () => {
  const wrapper = shallow(<Post title='preview' slug='preview' content='preview-content' excerpt='preview-excerpt'/>)
  const tree = toJson(wrapper)
  expect(tree).toMatchSnapshot()
})

it('should render h2 and p with message completely and without title link', () => {
  const wrapper = shallow(<Post title='preview' slug='preview' content='preview-content' excerpt='preview-excerpt' details/>)
  const tree = toJson(wrapper)
  expect(tree).toMatchSnapshot()
})
