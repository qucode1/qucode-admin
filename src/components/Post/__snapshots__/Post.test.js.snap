// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render h2 and p with message as preview 1`] = `
<div>
  <h2
    className="title"
  >
    <Link
      className="link"
      replace={false}
      to="/posts/preview"
    >
      preview
    </Link>
  </h2>
  <p
    className="content"
  >
    preview-excerpt
  </p>
</div>
`;

exports[`should render h2 and p with message completely and without title link 1`] = `
<div>
  <h2
    className="title"
  >
    preview
  </h2>
  <p
    className="content"
  >
    preview-content
  </p>
  <Link
    className="link"
    replace={false}
    to="/posts"
  >
    Go back to Posts
  </Link>
</div>
`;
